mtp_api/apps/transaction/serializers.py

```diff
@@ -31,3 +35,48 @@ class Meta:
             'owner',
             'credited',
         )
+
+
+class CreateTransactionListSerializer(serializers.ListSerializer):
+
+    @transaction.atomic
+    def create(self, validated_data):
+        upload_counter = Transaction.objects.all().aggregate(
+            Max('upload_counter')
+        )['upload_counter__max'] or 1
```

Comment:
yes possible race condition but unlikely to happen IMHO so not worth spending time on it unless you can think of any real case when this could happen.

Also, still not sure we really need an `upload_counter`, probably we don't. We can have a chat this or next week maybe and decide?